catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java

```diff
@@ -212,6 +213,68 @@
 
   private String customBackgroundSlideout;
 
+  private List<String> editorAttributes = Collections.emptyList();
+  private List<String> requiredAttributes = Collections.emptyList();
+  private Map<String, Set<String>> attributeEnumMap = Collections.emptyMap();
+
+  public List<String> getEditorAttributes() {
+    return editorAttributes;
+  }
+
+  public void setEditorAttributes(List<String> editorAttributes) {
+    this.editorAttributes = editorAttributes;
+  }
+
+  public List<String> getRequiredAttributes() {
+    return requiredAttributes;
+  }
+
+  public void setRequiredAttributes(List<String> requiredAttributes) {
+    this.requiredAttributes = requiredAttributes;
+  }
+
+  public Map<String, Set<String>> getAttributeEnumMap() {
+    return attributeEnumMap;
+  }
+
+  public void setAttributeEnumMap(Map<String, Set<String>> attributeEnumMap) {
+    this.attributeEnumMap = attributeEnumMap;
+  }
+
+  public void setAttributeEnumMap(List<String> entries) {
+    List<String> displayedAttributes = new ArrayList<>();
+    Map<String, Set<String>> attributeValueEnums = new HashMap<>();
+
+    for (String entry : entries) {
+      if (entry.isEmpty()) {
+        continue;
+      }
+
+      String[] kvPair = entry.split("=", 2);
+      String attrName = kvPair[0];
+      displayedAttributes.add(attrName);
+
+      if (kvPair.length == 2) {
+        Set<String> valueSet =
+            attributeValueEnums.containsKey(attrName)
+                ? attributeValueEnums.get(attrName)
+                : new LinkedHashSet<>();
+        for (String value : kvPair[1].split(",")) {
+          String trimmedValue = value.trim();
+          if (trimmedValue.length() > 0) {
+            valueSet.add(value.trim());
+          }
+        }
+        if (!valueSet.isEmpty()) {
+          attributeValueEnums.put(attrName, valueSet);
+        }
+      }
+    }
+
+    setEditorAttributes(displayedAttributes);
+    setAttributeEnumMap(attributeValueEnums);
```

Comment:
✏️ Totally not necessary, but to make this easier to follow I would just do `this.attributeEnumMap = attributeValueEnums`

and then I would get rid of all the `EditorAttributes` configs in this file and just use AttributeEnumMap.keySet where you need it